fix inspect format result

hqhq opened the pull request:
Currently `docker inspect -f` use json.Unmarshal() unmarshal
to interface, it will store all JSON numbers in float64, so
we use `docker inspect 4f0d73b75a0d | grep Memory` and
`docker inspect -f {{.HostConfig.Memory}} 4f0d73b75a0d` will
get different values.

hqhq commented:
ping

cpuguy83 commented on integration-cli/docker_cli_inspect_test.go:
```diff
@@ -21,3 +21,27 @@ func TestInspectImage(t *testing.T) {
 	logDone("inspect - inspect an image")
 }
+
+func TestInspectInt64(t *testing.T) {
```
This needs to be updated to use the new go-check suite.

cpuguy83 commented on integration-cli/docker_cli_inspect_test.go:
```diff
@@ -21,3 +21,27 @@ func TestInspectImage(t *testing.T) {
 }
+
+func TestInspectInt64(t *testing.T) {
+	defer deleteAllContainers()
```
Won't need this after switching to use go-check

cpuguy83 commented:
Needs a rebase.
Also, I could have sworn I saw a fix for this a few weeks ago... maybe it was in a different part of the codebase?

runcom commented:
@cpuguy83 this https://github.com/docker/docker/pull/11839 ?

cpuguy83 commented:
Nah, it was something specifically setting the decoder to use numbers.... but anyway this is definitely needed, imo.

LK4D4 commented:
@hqhq Need rebase, sorry :(

hqhq committed:
fix inspect format result

Currently `docker inspect -f` use json.Unmarshal() unmarshal
to interface, it will store all JSON numbers in float64, so
we use `docker inspect 4f0d73b75a0d | grep Memory` and
`docker inspect -f {{.HostConfig.Memory}} 4f0d73b75a0d` will
get different values.

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

hqhq commented:
@cpuguy83 @LK4D4 Rebased.

cpuguy83 commented:
LGTM

duglin commented:
please squash your commits

hqhq commented:
@duglin Done.

duglin commented:
thanks!  LGTM once janky is done
